### DISCUSSION TITLE
Get a list of provider-specific literals used in text commands.

### DISCUSSION THREAD

**Discussion by acardenaskma** [dotnet/SqlClient#1888] [Q&A]
Can we get a list of provider-specific literals used in text commands, using SqlConnection?

**Comment by lcheunglci** [comment]
Are you asking for what are the supported keywords used in the connection string?  If so, you can use the names of the [properties](https://learn.microsoft.com/en-us/dotnet/api/microsoft.data.sqlclient.sqlconnectionstringbuilder?view=sqlclient-dotnet-standard-5.0#properties) used in the [SqlConnectionStringBuilder](https://learn.microsoft.com/en-us/dotnet/api/microsoft.data.sqlclient.sqlconnectionstringbuilder) unless you're referring to the [SqlCommand.CommandText](https://learn.microsoft.com/en-us/dotnet/api/microsoft.data.sqlclient.sqlcommand.commandtext?view=sqlclient-dotnet-standard-5.0#microsoft-data-sqlclient-sqlcommand-commandtext), in that case, could you provide an example?

**Comment by acardenaskma** [comment]
Hello, 
Yes, I'm referring to the literals used in the command text.
More here: [oledbschemaguid.dbinfoliterals](https://learn.microsoft.com/en-us/dotnet/api/system.data.oledb.oledbschemaguid.dbinfoliterals)

Here's an example of how to get the literals using an ADODB Connection
```
        public static void LoadSchemaLiterals()
        {
            var connectionString = "....SQL Server connection string...";
            ADODB.Connection connection = new ADODB.Connection();
            connection.Open(connectionString, null, null);
            ADODB.Recordset recordSet = connection.OpenSchema(ADODB.SchemaEnum.adSchemaDBInfoLiterals, null, null);

            /*
            This will return records listing the schema literals from the database

            "QUOTE";
            "QUOTE_SUFFIX"
            "LIKE_PERCENT"
            "ESCAPE_PERCENT"
            "ESCAPE_PERCENT_SUFFIX"
            "LIKE_UNDERSCORE"
            "ESCAPE_UNDERSCORE"
            "ESCAPE_UNDERSCORE_SUFFIX"

            */
        }
```
